### DISCUSSION TITLE
Entity listeners: what's the difference between passed object and event arguments object

### DISCUSSION THREAD

**Discussion by ToshY** [doctrine/doctrine2#10922] [Support Questions] [answered]
I'm looking at the documentation regarding ["Entity Listeners"](https://www.doctrine-project.org/projects/doctrine-orm/en/2.16/reference/events.html#entity-listeners), and I see the following example for `preUpdate`:

```php
<?php
use Doctrine\ORM\Event\PreUpdateEventArgs;

class UserListener
{
    public function preUpdate(User $user, PreUpdateEventArgs $event)
    {
        // Do something on pre update.
    }
}
```

The explanation above that example denotes the following:

> An entity listener method receives two arguments, the entity instance and the lifecycle event.

My question:

* What's the difference between the first argument `$user` and if would get the object from the event arguments, `$event->getObject()` ?
    1. If there's a difference, what is it? 
    2. If there isn't difference, why is there a first argument?

**Comment by ghost** [comment, marked as answer]
Yes, they are the same. You can see it in `UnitOfWork::executeUpdates` where the event is dispatched or just check it yourself with `===` comparison. I don't know why this is that way but there might be no reason and this is still fine.

**Reply by ToshY** [reply]
Thank you for the confirmation.
